### PR TITLE
Update c++ SDK and fix network connections.

### PR DESCRIPTION
**Pull request by dcmiddle**
Network fixes From Jamie and some SDK updates from Dan.

### CODE REVIEW

**Comment by agunde406** [comment]
Tested locally

File: sdk/cxx/src/transaction_processor.cpp [hunk 1]
```diff
@@ -138,25 +140,80 @@ void TransactionProcessor::HandleProcessingRequest(const void* msg,
         LOG4CXX_ERROR(logger, "TransactionProcessor -> Apply error unknown");
         response.set_status(TpProcessResponse::INTERNAL_ERROR);
     }
-    this->response_stream->SendMessage(
-        Message_MessageType_TP_PROCESS_RESPONSE, response);
+    this->response_stream->SendResponseMessage(
+            Message_MessageType_TP_PROCESS_RESPONSE, response, correlation_id);
```
Comment (cianx) [inline sdk/cxx/src/transaction_processor.cpp]: remove extra indent.

File: sdk/examples/intkey_cxx/int_key_transaction_processor.cpp [hunk 2]
```diff
@@ -201,13 +201,13 @@ class IntKeyApplicator:  public sawtooth::TransactionApplicator {
...
                 std::stringstream error;
-                error << "Verb was 'dec', but value does not exist exists: " <<
+                error << "Verb was 'dec', but value does not exist for " <<
                         "Name: " << name;
                 throw sawtooth::InvalidTransaction(error.str());
             }
         } else {
             std::stringstream error;
-            error << "Verb was 'dec', but value does not exist exists: " <<
+            error << "Verb was 'dec', but address not found in state. " <<
```
[the remark this answers is not in the document]
Comment (dcmiddle) [reply sdk/examples/intkey_cxx/int_key_transaction_processor.cpp]: @cianx For these, I was explicitly labeling "Name" in the log statement so as not to imply the string was an address. Was the intent of the feedback to change the "." to ":" or drop that Name label? Same for the other above.

**Comment by dcmiddle** [comment]
@cianx I fixed per your feedback in separate commits for easier re-review. I'll re-rebase them in after you re-review.

**Pull request merged**